client/src/shared/components/Dropdown/index.tsx

```diff
@@ -1,121 +1,76 @@
-/* eslint-disable @typescript-eslint/strict-boolean-expressions */
-import React, { useState, useRef, useEffect, type ReactElement } from 'react';
-import ChevronDownIcon from '@/src/shared/icons/ChevronDownIcon';
-import { useOutsideClick } from '@/src/shared/hooks/useOutsideClick';
-import { useSignOut } from '@/src/shared/hooks/useSignOut';
-import Link from 'next/link';
+import React, { useState } from 'react';
 
-export interface Option {
-  text: string;
-  url: string;
+export interface SortOption {
+  label: string;
+  value: string;
+  icon?: React.ReactNode;
 }
 
 export interface DropdownProps {
-  options: Option[];
-  classNames?: string;
-  showLogoutButton?: boolean;
-  children?: ReactElement;
-  showArrowIcon?: boolean;
+  buttonText: string;
+  buttonIcon?: React.ReactNode;
+  isEditMode?: boolean;
```

Comment:
Now that it is a reusable dropdown component, please give names of props that relates to dropdown, `isEditMode` might be confusing by other devs, depending on where they are using it, instead please change it to `disabled`, since that is what is it used for down below